src/filter.js

```diff
@@ -0,0 +1,105 @@
+var Crossfilter = require("crossfilter");
+
+function Filter(){
+  this.crossfilter = new Crossfilter();
+  this.dimensions = {};
+  this.tiles = {};
+  this.report = {};
+  this.expressions = {};
+};
+
+Filter.prototype = {
+  addTile: function(tilePoint, collection){
+    var tilePointString = tilePoint.zoom + ":" + tilePoint.x + ":" + tilePoint.y;
+    if (typeof this.tiles[tilePointString] != 'undefined') return this.getTile(tilePoint);
+    this.crossfilter.add(collection.features.map(function(f){
+      f.properties.tilePoint = tilePoint.zoom + ":" + tilePoint.x + ":" + tilePoint.y;
+      return f;
+    }));
+    this.tiles[tilePointString] = true;
+    return this.getTile(tilePoint);
+  },
+
+  removeTile: function(tilePoint){
+    var tilePointString = tilePoint.zoom + ":" + tilePoint.x + ":" + tilePoint.y;
+    if(!this.dimensions.tiles){
+      return;
+    }
+    this.dimensions.tiles.filter(tilePointString);
+    this.crossfilter.remove();
+    this.dimensions.tiles.filterAll();
+    delete this.tiles[tilePointString];
+  },
+
+  getTile: function(tilePoint){
+    var tilePointString = tilePoint.zoom + ":" + tilePoint.x + ":" + tilePoint.y;
+    if(!this.dimensions.tiles){
+      this.dimensions.tiles = this.crossfilter.dimension(function(f){return f.properties.tilePoint});
+    }
+    var tile = {type: "FeatureCollection", features: null};
+    this.dimensions.tiles.filter(tilePointString);
+    tile.features = this.dimensions.tiles.top(Infinity);
+    this.dimensions.tiles.filterAll();
+    return tile;
+  },
+
+  filterRange: function(column, range){
+    if (!this.dimensions[column]){
+      this.dimensions[column] = this.crossfilter.dimension(function(f){return f.properties[column]});
+    }
+    this.dimensions[column].filter(range);
+
+  },
+
+  filterAccept: function(column, terms){
+    if (!this.dimensions[column]){
+      this.dimensions[column] = this.crossfilter.dimension(function(f){return f.properties[column]});
+    }
+    if (!(terms instanceof Array)){
+      terms = [terms];
+    }
+    this.dimensions[column].filter(function(f){
+      if (terms.indexOf(f) > -1){
+        return true;
+      }
+    });
+
+  },
+
+  filterReject: function(column, terms){
```

Comment:
All filter* functions are quite similar. Wouldn't be better to have a generic filter function that receives the actual filter function and applies it by column?

Or even better, having different filter entities with a common interface that can be applied. Having different functions for different types of filters means that we need as many public functions as filters we want to have.